packages/special-pages/pages/duckplayer/src/js/index.js

```diff
@@ -86,10 +87,42 @@ const VideoPlayer = {
      * Sets up the video player:
      * 1. Fetches the video id
      * 2. If the video id is correctly formatted, it loads the YouTube video in the iframe, otherwise displays an error message
+     * @param {object} opts
+     * @param {string} opts.base
      */
-    init: () => {
+    init: (opts) => {
```

Comment:
This singleton pattern is a little odd at this point. We're holding state in an instance, it'd be much simpler to be a class you construct with the state.

Obviously not blocking on this at all, just wanted to state that this is becoming a bit of a smell if we're going to do more in this direction.

Reply:
Agreed yeah - it's from the original port and luckily isolated to this page for now. but we should improve for sure :)